lib/masamune/abstract_syntax_tree.rb

```diff
@@ -11,145 +11,86 @@ def initialize(code)
         LexNode.new(raw_lex_nodes.index(lex_node), lex_node, self.__id__)
       end
 
-      @node_list = []
-      @data_node_list = []
-      @comment_list = []
-      register_nodes(@tree)
-
-      # Refer to Masamune::AbstractSyntaxTree::Comment
-      # to see why we register these separately.
-      register_comments
+      @prism = Prism.parse(code)
+      @nodes = []
+      register_nodes
+      @nodes = order_nodes(@nodes)
     end
 
-    def register_nodes(tree_node = self.tree)
-      if tree_node.is_a?(Array)
-        klass = get_node_class(tree_node.first)
-        msmn_node = klass.new(tree_node, self.__id__)
-      else
-        # Create a general node if the node is a single value.
-        msmn_node = Node.new(tree_node, self.__id__)
-      end
-
-      # Register nodes and any data nodes housed within it.
-      # See Masamune::AbstractSyntaxTree::DataNode for more details on what a data node is.
-      @node_list << msmn_node
-      msmn_node.data_nodes.each { |dn| @data_node_list << dn } if msmn_node.data_nodes
+    def register_nodes(tree_node = @prism.value)
+      # Include our in-house Masamune node helper for all Prism nodes present.
+      tree_node.class.include(NodeHelper) unless tree_node.class.include?(NodeHelper)
```

Comment:
Doing this dynamically every time you parse a tree is going to be detrimental to performance and caching. Probably easier to put

```ruby
Prism::Node.include(NodeHelper)
```

at this top of this file instead

Reply:
Resolved in #61.
We ended up extending the `Prism::Node` class directly and moving the methods from `NodeHelper` to `lib/masamune/abstract_syntax_tree/prism/node_extensions.rb`.
cc/ @kaspth